frontend/pages/component/NavBar.tsx

```diff
@@ -19,7 +23,11 @@ const navigation = [
 
 export default function NavBar() {
     const [sidebarOpen, setSidebarOpen] = useState(false);
-
+    const [darkTheme, setDarkTheme] = useState(isDarkTheme());
```

Comment:
Why define an useState to save if theme dark is used when you already have a function for that ?
You even use this function to set the default value